src/util/std_types.h

```diff
@@ -890,6 +890,21 @@ class code_typet:public typet
       result.push_back(it->get_identifier());
     return result;
   }
+
+  typedef
+    std::unordered_map<irep_idt, std::size_t, irep_id_hash> parameter_indicest;
+
+  /// Get a map from parameter name to its index
+  parameter_indicest parameter_indices() const
+  {
+    parameter_indicest parameter_indices;
+    const parameterst &p = parameters();
+    parameter_indices.reserve(p.size());
+    std::size_t i = 0;
+    for(const auto &parameter : parameters())
+      parameter_indices.insert({ parameter.get_identifier(), i++ });
```

Comment:
Parameters need not have an identifier, in which case the results will at best cause confusion. Thus `parameter_identifiers` exists in `goto_function_templatet`.